vehicle_status_generator/src/vehicle_status_generator_worker.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2019-2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "vehicle_status_generator_worker.h"
+#include <stdlib.h>
+#include <ros/ros.h>
+#include <random>
+
+namespace vehicle_status_generator
+{
+    VehicleStatusGeneratorWorker::VehicleStatusGeneratorWorker() {}
+    
+    uint8_t VehicleStatusGeneratorWorker::getNextMsgCount()
+    {
+        uint8_t old_msg_count = msg_count_;
+        msg_count_++;
+        if(msg_count_ == 128)
```

Comment:
See comment below, but for the record the cleanest way to implement this would be with the modulus operator ` msg_count = ++msg_count % 128` handles the rollover cleanly.

This also only ever returns the previous message count despite the method name being `getNext...` which is slightly confusing, but not a big deal functionally.